### PR TITLE
Issue/324 manger test

### LINKED ISSUE #324: Document new architecture (including a diagram)
Once PR #307 is merged, add some documentation (to README.md?) to describe how `prepare_ml_data.py` works.

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

add tests for manager
- test that checks batches are made
- test that checks config is saved

Want to do this before #341 

Fixes #324 

## How Has This Been Tested?

normal unit test
+ addition ones for this PR

- [ ] No
- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: tests/test_manager.py [hunk 1]
```diff
@@ -77,3 +79,83 @@ def test_get_daylight_datetime_index():
 
 
 # TODO: Issue #322: Test the other Manager methods!
+
```
Comment (peterdudfield) [inline tests/test_manager.py:82]: remove TODO

**Comment by peterdudfield** [comment]
Tests pass - locally - strange ....

File: nowcasting_dataset/data_sources/data_source.py [hunk 2]
```diff
@@ -182,6 +182,8 @@ def create_batches(
         # Split locations per example into batches:
         n_batches = len(spatial_and_temporal_locations_of_each_example) // batch_size
         locations_for_batches = []
+        logger.warning("xxxxx")
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/data_source.py:185]: Is this warning still required?  :slightly_smiling_face:

File: nowcasting_dataset/data_sources/data_source.py [hunk 3]
```diff
@@ -182,6 +182,8 @@ def create_batches(
...
+        logger.warning(n_batches)
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/data_source.py:186]: Please make this warning a bit more verbose?

File: nowcasting_dataset/data_sources/data_source.py [hunk 4]
```diff
@@ -193,7 +195,7 @@ def create_batches(
         # Loop round each batch:
         for n_batches_processed, locations_for_batch in enumerate(locations_for_batches):
             batch_idx = idx_of_first_batch + n_batches_processed
-            logger.debug(f"{self.__class__.__name__} creating batch {batch_idx}!")
+            logger.warning(f"{self.__class__.__name__} creating batch {batch_idx}!")
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/data_source.py:198]: Sorry if I've misunderstood but why does this message need to be a warning?

File: nowcasting_dataset/data_sources/data_source.py [hunk 5]
```diff
@@ -205,6 +207,7 @@ def create_batches(
             # Save batch to disk.
             netcdf_filename = path_to_write_to / nd_utils.get_netcdf_filename(batch_idx)
             batch.to_netcdf(netcdf_filename)
+            logger.warning(f"Save file to  {netcdf_filename}")
```
Comment (JackKelly) [inline nowcasting_dataset/data_sources/data_source.py:210]: Same as above:  Does this message really need to be a warning?

File: nowcasting_dataset/filesystem/utils.py [hunk 6]
```diff
@@ -10,15 +10,24 @@
...
-def upload_and_delete_local_files(dst_path: str, local_path: Path):
+def upload_and_delete_local_files(dst_path: Union[str, Path], local_path: Union[str, Path]):
     """
     Upload an entire folder and delete local files to either AWS or GCP
     """
     _LOG.info("Uploading!")
     filesystem = get_filesystem(dst_path)
-    filesystem.put(str(local_path), dst_path, recursive=True)
+
+    _LOG.warning(f"moving files from {local_path} to {dst_path}")
```
Comment (JackKelly) [inline nowcasting_dataset/filesystem/utils.py:20]: Does this need to be a warning?

File: nowcasting_dataset/filesystem/utils.py [hunk 7]
```diff
@@ -10,15 +10,24 @@
...
+
+    _LOG.warning(get_all_filenames_in_path(local_path))
```
Comment (JackKelly) [inline nowcasting_dataset/filesystem/utils.py:22]: Please make these two messages a bit more verbose  (or maybe remove these two messages?)  Also, do these need to be warnings?

File: nowcasting_dataset/filesystem/utils.py [hunk 8]
```diff
@@ -10,15 +10,24 @@
...
+    _LOG.warning(get_all_filenames_in_path(dst_path))
+
+    filesystem.put(str(local_path) + "/", str(dst_path) + "/", recursive=True)
     delete_all_files_in_temp_path(local_path)
 
+    _LOG.warning(get_all_filenames_in_path(local_path))
```
Comment (JackKelly) [inline nowcasting_dataset/filesystem/utils.py:28]: Same as above :)

File: nowcasting_dataset/manager.py [hunk 9]
```diff
@@ -389,18 +397,25 @@ def create_batches(self, overwrite_batches: bool) -> None:
...
+                        logger.warning(
+                            f"Not saving uploading batches so have not made {local_temp_path}"
+                        )
 
                     # Submit data_source.create_batches task to the worker process.
-                    future = executor.submit(
-                        data_source.create_batches,
+                    print(executor)
+                    # future = executor.submit(
+                    logger.warning("Making batches")
```
Comment (JackKelly) [inline nowcasting_dataset/manager.py:408]: Do these need to be warnings?

File: requirements.txt [hunk 10]
```diff
@@ -21,5 +21,5 @@ plotly
 tqdm
 black
 pre-commit
-fsspec
+fsspec==2021.7.0
```
Comment (JackKelly) [inline requirements.txt:24]: Ooh, interesting, are we sure we can't use more recent versions of `fsspec`?  A few days ago, there was a bug (I think) in fsspec which forced pip to install an ancient version of `gscfs` but that appears fixed now (if that's what make you pin `fsspec` to version 2021.7.0?

**Review by JackKelly — COMMENTED** [review]
Looks good!

A few minor comments...

If you're using `logger.warning` so the log messages appear during unittests then I think there is a way to set the log level to, say, debug during tests (but I can't remember exactly how to do that!)

**Comment by peterdudfield** [comment]
Hey @JackKelly sorry i closed this PR and move to https://github.com/openclimatefix/nowcasting_dataset/pull/363

**Pull request closed without merging**